### PR TITLE
Fix null refs in byte array COM interop

### PR DESCRIPTION
**Pull request by jtschuster**
Roslyn API doesn't use nullable types and will return null on properties and fields that aren't annotated with the `?`. The linker assumes that fields will not be null, and those issues tend to appear most frequently with COM interop when a type doesn't have a namespace.

This fixes the null reference in a minimal repro, but there are likely more holes to address https://github.com/dotnet/linker/issues/2687.

This should fix https://github.com/dotnet/linker/issues/2686.

I wasn't sure where the test case best fits in, so it's in a separate file in the `Interop` test cases folder. Let me know if there's a better place for it.

### CODE REVIEW

File: test/Mono.Linker.Tests.Cases/Interop/ByteArrayCom.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.Interop
+{
+	[SkipKeptItemsValidation]
+	class ByteArrayCom
```
Comment (vitek-karas) [inline test/Mono.Linker.Tests.Cases/Interop/ByteArrayCom.cs:15]: ExpectNoWarnings

File: src/ILLink.RoslynAnalyzer/COMAnalyzer.cs [hunk 2]
```diff
@@ -88,11 +88,11 @@ static bool IsComInterop (ISymbol symbol)
 				if (typeSymbol == null)
 					return false;
 
-				if (typeSymbol.ContainingNamespace.Name == "System" && typeSymbol.Name == "Array") {
+				if (typeSymbol.ContainingNamespace?.Name == "System" && typeSymbol.Name == "Array") {
```
Comment (marek-safar) [inline src/ILLink.RoslynAnalyzer/COMAnalyzer.cs:91]: Don't we have enabled nullable to catch such errors?

Comment (vitek-karas) [reply src/ILLink.RoslynAnalyzer/COMAnalyzer.cs:91]: See https://github.com/dotnet/linker/issues/2687.

**Comment by jtschuster** [comment]
The issues were fixed by https://github.com/dotnet/linker/pull/2692, this now just adds tests.

**Review by tlakollo — APPROVED** [review]

**Review by vitek-karas — APPROVED** [review]

**Comment by eerhardt** [comment]
For my education, how does adding new tests fix the original issue? Where is the source code change that fixed #2686?

**Comment by akoeplinger** [comment]
@eerhardt see https://github.com/dotnet/linker/pull/2688#issuecomment-1072774547

**Pull request merged**